failed to build with ESP IDF

the error is:

```
/home/bartolomey/projects/aether-client-cpp/third_party/stdexec/include/stdexec/__detail/__stop_when.hpp:107:27: error: expected identifier before numeric constant
  107 |           template <class _C>
      |                           ^~
/home/bartolomey/projects/aether-client-cpp/third_party/stdexec/include/stdexec/__detail/__stop_when.hpp:107:27: error: expected '>' before numeric constant
In file included from /home/bartolomey/projects/aether-client-cpp/third_party/stdexec/include/stdexec/__detail/__counting_scopes.hpp:28,
                 from /home/bartolomey/projects/aether-client-cpp/third_party/stdexec/include/stdexec/execution.hpp:29,
                 from /home/bartolomey/projects/aether-client-cpp/aether/executors/executors.h:22,
                 from /home/bartolomey/projects/aether-client-cpp/aether/all.h:27,
                 from /home/bartolomey/projects/aether-client-cpp/examples/cloud/cloud_test.cpp:20:
/home/bartolomey/projects/aether-client-cpp/third_party/stdexec/include/stdexec/__detail/__stop_when.hpp:108:29: error: type/value mismatch at argument 2 in template parameter list for 'template<class _Ty, class ... _As> concept stdexec::__std::constructible_from'
  108 |             requires __std::constructible_from<_Fn, _C>
      |                             ^~~~~~~~~~~~~~~~~~~~~~~~~~~
```

... and so on.

The problem is that `_C` is defined as macro in ctype.h on that platform.

I found the same issue https://github.com/kokkos/mdspan/issues/187

Also clang-tidy complains on that code with https://clang.llvm.org/extra/clang-tidy/checks/bugprone/reserved-identifier.html. I not sure if that applicable for stdexec as it's reference for standard implementation, but why use `_FN` `_C` instead of just `FN` and `C`?

--- a/include/stdexec/__detail/__parallel_scheduler.hpp
+++ b/include/stdexec/__detail/__parallel_scheduler.hpp
@@ -62,7 +62,7 @@ namespace STDEXEC
   class parallel_scheduler;
   class __parallel_sender;
 
-  template <bool, sender _S, std::integral _Size, class _Fn, bool>
+  template <bool, sender _Sender, std::integral _Size, class _Fn, bool>
   class __parallel_bulk_sender;
 
   /// Returns a scheduler that can add work to the underlying execution context.

--- a/include/stdexec/__detail/__stop_when.hpp
+++ b/include/stdexec/__detail/__stop_when.hpp
@@ -104,20 +104,20 @@ namespace STDEXEC
         template <class _Fn>
         struct callback_type
         {
-          template <class _C>
-            requires __std::constructible_from<_Fn, _C>
-          explicit callback_type(__fused_token&& __ftkn, _C&& __fn)
-            noexcept(__nothrow_constructible_from<_Fn, _C>)
-            : __fn_(static_cast<_C&&>(__fn))
+          template <class _Cb>
+            requires __std::constructible_from<_Fn, _Cb>
+          explicit callback_type(__fused_token&& __ftkn, _Cb&& __fn)
+            noexcept(__nothrow_constructible_from<_Fn, _Cb>)
+            : __fn_(static_cast<_Cb&&>(__fn))
             , __cb1_(std::move(__ftkn.__tkn1_), __cb(this))
             , __cb2_(std::move(__ftkn.__tkn2_), __cb(this))
           {}
 
-          template <class _C>
-            requires __std::constructible_from<_Fn, _C>
-          explicit callback_type(__fused_token const & __ftkn, _C&& __fn)
-            noexcept(__nothrow_constructible_from<_Fn, _C>)
-            : __fn_(static_cast<_C&&>(__fn))
+          template <class _Cb>
+            requires __std::constructible_from<_Fn, _Cb>
+          explicit callback_type(__fused_token const & __ftkn, _Cb&& __fn)
+            noexcept(__nothrow_constructible_from<_Fn, _Cb>)
+            : __fn_(static_cast<_Cb&&>(__fn))
             , __cb1_(__ftkn.__tkn1_, __cb(this))
             , __cb2_(__ftkn.__tkn2_, __cb(this))
           {}

